consensus/types/src/runtime_var_list.rs

```diff
@@ -217,6 +218,21 @@ where
     }
 }
 
+impl<'de, C> ContextDeserialize<'de, ForkName> for RuntimeVariableList<C>
+where
+    C: ContextDeserialize<'de, ForkName>,
+{
+    fn context_deserialize<D>(deserializer: D, context: ForkName) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        // first parse out a Vec<C> using the Vec<C> impl you already have
+        let vec: Vec<C> = Vec::context_deserialize(deserializer, context)?;
+        let len = vec.len();
+        Ok(RuntimeVariableList::from_vec(vec, len))
```

Comment:
I actually think this is a use case for `ContextDeserialize<'de, usize>` if we can make that work.

I don't think it makes sense to deserialize an arbitrary length `RuntimeVariableList` (i.e. using `vec.len()` as `max_len`), because the whole point of `RuntimeVariableList` is that we provide a runtime bound and check it.

I can see that removing this impl breaks the HTTP client code for `get_blobs`. Maybe we could add the `ContextDeserialize<'de, usize>` impl and try to use that there instead. It may require fussing around with `ExecutionOptimisticFinalizedBeaconResponse` though in order to also parse the metadata. We kind of just want the `ChainSpec` as part of the context as well so we can get the right blob bound.

Reply:
I've tried to implement this in [`64ab14c`](https://github.com/sigp/lighthouse/pull/7372/commits/64ab14cfb0edb495c9c35d9cd647a296e6443d7f). Let me know what you think!

Reply:
I like!